.goreleaser.yml

```diff
@@ -1,131 +1,56 @@
-env:
-  - CGO_ENABLED=1
 builds:
-  - id: sat-linux-amd64
-    main: .
-    binary: sat
-    env:
-      - CC=x86_64-linux-gnu-gcc
-      - CXX=x86_64-linux-gnu-g++
-    goos:
-      - linux
-    goarch:
-      - amd64
-    ldflags:
-      - -s -w -X main.version={{.Version}} -X main.commit={{.Commit}} -X main.date={{.Date}}
-      - -extldflags "-static"
-    tags:
-      - netgo
-      - wasmtime
-  - id: sat-linux-arm64
-    main: .
-    binary: sat
-    env:
-      - CC=aarch64-linux-gnu-gcc
-      - CXX=aarch64-linux-gnu-g++
-    goos:
-      - linux
-    goarch:
-      - arm64
-    ldflags:
-      - -s -w -X main.version={{.Version}} -X main.commit={{.Commit}} -X main.date={{.Date}}
-      - -extldflags "-static"
-    tags:
-      - netgo
-      - wasmtime
-  - id: sat-darwin-amd64
-    main: .
-    binary: sat
-    env:
-      - CC=o64-clang
-      - CXX=o64-clang++
-    goos:
-      - darwin
-    goarch:
-      - amd64
-    ldflags:
-      - -s -w -X main.version={{.Version}} -X main.commit={{.Commit}} -X main.date={{.Date}}
-    tags:
-      - netgo
-      - wasmtime
-  - id: sat-darwin-arm64
-    main: .
-    binary: sat
-    env:
-      - CC=oa64-clang
-      - CXX=oa64-clang++
-    goos:
-      - darwin
-    goarch:
-      - arm64
-    ldflags:
-      - -s -w -X main.version={{.Version}} -X main.commit={{.Commit}} -X main.date={{.Date}}
-    tags:
-      - netgo
-      - wasmtime
-
-  - id: constd-linux-amd64
-    main: ./constd
-    binary: constd
-    env:
-      - CC=x86_64-linux-gnu-gcc
-      - CXX=x86_64-linux-gnu-g++
-    goos:
-      - linux
-    goarch:
-      - amd64
-    ldflags:
-      - -s -w -X main.version={{.Version}} -X main.commit={{.Commit}} -X main.date={{.Date}}
-      - -extldflags "-static"
-    tags:
-      - netgo
-      - wasmtime
-  - id: constd-linux-arm64
-    main: ./constd
-    binary: constd
-    env:
-      - CC=aarch64-linux-gnu-gcc
-      - CXX=aarch64-linux-gnu-g++
-    goos:
-      - linux
-    goarch:
-      - arm64
-    ldflags:
-      - -s -w -X main.version={{.Version}} -X main.commit={{.Commit}} -X main.date={{.Date}}
-      - -extldflags "-static"
-    tags:
-      - netgo
-      - wasmtime
-  - id: constd-darwin-amd64
-    main: ./constd
-    binary: constd
-    env:
-      - CC=o64-clang
-      - CXX=o64-clang++
-    goos:
-      - darwin
-    goarch:
-      - amd64
-    ldflags:
-      - -s -w -X main.version={{.Version}} -X main.commit={{.Commit}} -X main.date={{.Date}}
-    tags:
-      - netgo
-      - wasmtime
-  - id: constd-darwin-arm64
+  - <<: &build_defaults
```

Comment:
Are they alias nodes? https://yaml.org/spec/1.2.2/#71-alias-nodes

What's the `<<:` for?

Reply:
You're right, `build_defaults` is a variable (anchor) to reused.

`<<` is a merge key thing. Because we're storing an object (map, dictionary, etc) as an anchor, we want to properly merge it in. Spec is here: https://yaml.org/type/merge.html